### ISSUE TITLE
tensorboard not updating

### ISSUE THREAD

**Issue by xyoungli** [pytorch/pytorch#20477]
i wrote a loss summary every 10 iters, while tensorboard did not show the latest loss.
eg. training iter now is 1000, while tensorboard only shows 100 
following is my code:
initialization:
        self.writer = SummaryWriter(log_dir=cfg.LOG_DIR, flush_secs=10)
loss:
        self.writer.add_scalar('Train/iter_loss', losses.val.item(), i + (epoch - 1) * epoch_size)

**Comment by seenitall** [comment]
Same here, I need to kill the tensorboard session and start it again to get it updated (page reloading does not help -- Firefox).

**Comment by pietern** [comment]
cc @orionr

**Comment by orionr** [comment]
cc @lanpa - any thoughts? Also, cc @natalialunova as an FYI

**Comment by A1R0B0T** [comment]
I have the same problem

**Comment by lanpa** [comment]
@xyoungli Does @seenitall 's method work for you?
Which version of tensorboard are you using?

**Comment by xyoungli** [comment]
> @xyoungli Does @seenitall 's method work for you?
> Which version of tensorboard are you using?

tb-nightly, 1.14.0-20190513
it works, but that's not the normal way to update

**Comment by rotmanmi** [comment]
My current workaround:

while true; do
        timeout -sHUP 1m tensorboard --logdir=runs;
done

**Comment by SunDoge** [comment]
tb-nightly           1.14.0a20190603
Still fail to update automatically.

**Comment by odysseus0** [comment]
Same problem here. Any prospect on a future fix?

**Comment by odysseus0** [comment]
Guys, downgrading TensorBoard to 1.13.1 works for me.

**Comment by orionr** [comment]
@nfelt any chance you'd have thoughts here too? We'll investigate - thank you for the details.

**Comment by nfelt** [comment]
If killing TensorBoard and restarting fixes this, it's probably because the event file being written to is not the most recent one in the log directory, and TensorBoard currently reads new data only from the last event file in each directory (it reads each one all the way during the first pass, but then reads only from the last file on subsequent passes).

You can avoid this by only using one SummaryWriter per log directory.

If that doesn't sound like the issue (e.g. you only have a single event file in your directory), please comment with the details of your log directory structure and ideally the full output of running `tensorboard --verbosity 2` and having the problem occur.

**Comment by neherh** [comment]
Here is an output of the error which isn't @nfelt's initial suggestion:
[errorfile.txt](https://github.com/pytorch/pytorch/files/3263405/errorfile.txt)

I modified a version of https://github.com/lanpa/tensorboardX with added time.sleep(0.5) to generate the points at a slower speed and the tensorboard was not able to update. Code Structure to write the scale value looks likes this:

    writer.add_scalar('data/scalar1', dummy_s1[0], n_iter)
    writer.add_scalar('data/scalar2', dummy_s2[0], n_iter)

    writer.add_scalars('data/scalar_group', {'xsinx': n_iter * np.sin(n_iter),
                                             'xcosx': n_iter * np.cos(n_iter),
                                             'arctanx': np.arctan(n_iter)}, n_iter)

Actual directory structure looks like this:
/runs/events.out.tfevents.1559853561.ai-machine.18673.0
/runs/data/scalar_group/arctanx/events.out.tfevents.1559853562.ai-machine.18673.3
/runs/data/scalar_group/xcosx/events.out.tfevents.1559853562.ai-machine.18673.2
/runs/data/scalar_group/xsinx/events.out.tfevents.1559853562.ai-machine.18673.1

Tensorboard version: 1.14.0a20190529


It seems like to me these lines are suspect:
I0606 16:38:38.021887 140298998564608 event_file_loader.py:58] Loading events from ... /
I0606 16:38:38.802124 140298998564608 event_file_loader.py:73] Cannot read more events: No more events to read
I0606 16:38:38.802356 140298998564608 event_file_loader.py:79] No more events in

**Comment by nfelt** [comment]
@neherh the logs you posted include lines like

```
Directory watcher advancing from /home/hneher/Documents/gitRepos/pytorch-mnist-tensorboard-example/runs/events.out.tfevents.1559853104.ai-machine.18136.0 to /home/hneher/Documents/gitRepos/pytorch-mnist-tensorboard-example/runs/events.out.tfevents.1559853228.ai-machine.18313.0
```

This does indicate that there's more than one event file in per directory, in which case it could be the case that new data in the earlier file is being skipped.

If you have a self-contained script that reproduces the issue that would be helpful for taking a further look, but as is I'm not sure that this is a (new) issue on the tensorboard reading side.

**Comment by jj0mst** [comment]
Had the same problem.
Today I uninstalled tb-nightly, installed full **tf-nightly** and now my dashboard is updated properly (didn't even had to restart the writing process, just tensorboard).
So I guess that in the reduced feature set we get by installing just tb-nightly we are missing something.

**Comment by orionr** [comment]
This is helpful! Thank you @jj0mst.

**Comment by orionr** [comment]
I'm working on a fix for this. cc @natalialunova

**Comment by orionr** [comment]
Fix landed! Closing.

**Comment by lcswillems** [comment]
In which version is it available?

**Comment by orionr** [comment]
If that's the case it unfortunately sounds like this will be in TB 1.15 or a patch release. @nfelt do you know? Thanks.

**Comment by nfelt** [comment]
The fix went in after TB 1.14, so it's only in tb-nightly (which you can get by uninstalling tensorboard and then pip installing tb-nightly).

We do our major releases in sync with TF so the next one depends on timing for TF's big 2.0 release.  I afraid we're tight on bandwidth to do a patch release unless there's still a lot of affected users for whom tb-nightly isn't a good workaround.

**Comment by wangqin4377** [comment]
@jj0mst really thanks! I install TF instead TB alone, the problem gone!
